### PR TITLE
Add documentation for using SSE-C on S3 primary storage

### PR DESCRIPTION
**Pull request by juliushaertl**
In preparation for https://github.com/nextcloud/server/pull/32798

### CODE REVIEW

File: admin_manual/configuration_files/primary_storage.rst [hunk 1]
```diff
@@ -182,3 +182,39 @@ all files for that user in their corresponding bucket.
...
 `Nextcloud customer portal <https://portal.nextcloud.com/article/object-store-as-primary-storage-16.html>`_.
+
+
+------------------------
+SSE-C encryption support
+------------------------
+
+Nextcloud added support for server side encryption (also known as SSE-C) with AWS.
+
+This means Nextcloud can encrypt files such storage, using a client provided key, the moment they are uploaded. The decryption key is provided by the client. We are using the `SSE-C provided by AWS <http://docs.aws.amazon.com/AmazonS3/latest/dev/ServerSideEncryptionCustomerKeys.html>`_ and to do this for Amazon AWS, the client has to specify exact information, so these options we provide in Nextcloud config:
```
Comment (artonge) [inline admin_manual/configuration_files/primary_storage.rst:193]: I am not sure about that one.

- The files are encrypted on the bucket side, but it is not clear in `the moment they are uploaded.`
- Not sure about `a client provided key` and ` decryption key is provided by the client`. It seems that we are talking about two keys. But there is only one. Right ?
- This paragraph might go too much in the implementation details, and i don't think that an admin care about that

```suggestion
The encryption and decryption appens on the S3 bucket side with a key provided by the Nextcloud server.
```

File: admin_manual/configuration_files/primary_storage.rst [hunk 2]
```diff
@@ -182,3 +182,39 @@ all files for that user in their corresponding bucket.
 
 You can find out more information about upscaling with object storage and Nextcloud in the
 `Nextcloud customer portal <https://portal.nextcloud.com/article/object-store-as-primary-storage-16.html>`_.
+
+
+------------------------
+SSE-C encryption support
+------------------------
+
+Nextcloud added support for server side encryption (also known as SSE-C) with AWS.
```
Comment (artonge) [inline admin_manual/configuration_files/primary_storage.rst:191]:
```suggestion
Nextcloud supports server side encryption, also known as `SSE-C <http://docs.aws.amazon.com/AmazonS3/latest/dev/ServerSideEncryptionCustomerKeys.html>`, with compatible S3 bucket providers.
```

File: admin_manual/configuration_files/primary_storage.rst [hunk 3]
```diff
@@ -182,3 +182,39 @@ all files for that user in their corresponding bucket.
...
+
+This means Nextcloud can encrypt files such storage, using a client provided key, the moment they are uploaded. The decryption key is provided by the client. We are using the `SSE-C provided by AWS <http://docs.aws.amazon.com/AmazonS3/latest/dev/ServerSideEncryptionCustomerKeys.html>`_ and to do this for Amazon AWS, the client has to specify exact information, so these options we provide in Nextcloud config:
+
+The key can be specified with the :code:`sse_c_key` parameter which needs to be provided as a base64 encoded string with a maximum length of 32 bytes, if it does not meet the requirements. The key can be generated using the the following command:
```
Comment (artonge) [inline admin_manual/configuration_files/primary_storage.rst:195]: What does `if it does not meet the requirements` means in the sentence ?

```suggestion
You can specify the key by setting the :code:`sse_c_key` parameter in you config.php file. It needs to be provided as a base64 encoded string with a maximum length of 32 bytes.
The key can be generated using the the following command:
```

**Review by artonge — CHANGES_REQUESTED** [review]

**Comment by juliushaertl** [comment]
Thanks @artonge Good catches, adjusted that accordingly.

**Review by artonge — APPROVED** [review]
:ok_hand:

**Pull request merged**